app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    res_lst = {}
+
+    def inner(*args, **kwargs) -> None:
+        key = (args, tuple(sorted(kwargs.items())))
+        if key in res_lst:
+            print("Getting from cache")
+            return res_lst[key]
+        elif key not in res_lst:
+            res = func(*args, **kwargs)
+            res_lst[key] = res
+            print("Calculating new result")
+            return res
```

Comment:
```suggestion
        key = (args, tuple(sorted(kwargs.items())))
        if key in res_lst:
            print("Getting from cache")
        elif key not in res_lst:
            res_lst[key] = func(*args, **kwargs)
            print("Calculating new result")
        return res_lst[key]
```

Also, `res` and `res_lst` are not clear variable names, please use descriptive variable names.